### PR TITLE
Moved access_log_formatter to a new folder formatter

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

access_log_formatter is used by access_log and local_reply.  So it is renamed as
substitution_formatter. 
* create a new folder /source/common/substitution
* moved access_log_formatter to the new folder,

Risk Level: None
Testing:  unit-test

### CODE REVIEW

**Comment by qiwzhang** [comment]
@mattklein123 @htuch could you help to review it?

File: include/envoy/substitution/BUILD [hunk 1]
```diff
@@ -0,0 +1,19 @@
+licenses(["notice"])  # Apache 2
```
Comment (mattklein123) [inline include/envoy/substitution/BUILD:1]: How about include/envoy/formatter and source/common/formatter? I think that's a bit more general and still covers this case? Then we might be able to get the header formatter stuff in there also at some point?

Comment (qiwzhang) [reply include/envoy/substitution/BUILD:1]: OK, rename the folder as formatter, the file is still called substitution_formatter.h. Something like:

include/envoy/formatter/substitution_formatter.h
source/common/formatter/substition_formatter.{h,cc}  substitution_format_string.{h, cc}

Comment (qiwzhang) [reply include/envoy/substitution/BUILD]: Done

File: source/common/access_log/access_log_impl.cc [hunk 2]
```diff
@@ -11,10 +11,11 @@
 #include "envoy/runtime/runtime.h"
 #include "envoy/upstream/upstream.h"
 
-#include "common/access_log/access_log_formatter.h"
 #include "common/common/assert.h"
 #include "common/common/utility.h"
 #include "common/config/utility.h"
+
+//#include "common/formatter/substitution_formatter.h"
```
Comment (mattklein123) [inline source/common/access_log/access_log_impl.cc:18]: ?

Comment (qiwzhang) [reply source/common/access_log/access_log_impl.cc]: Done. Removed

**Pull request merged**